crates/line_ending/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+//! We maintain the invariant that all internal strings use `\n` as line separator.
+//! This module does line ending conversion and detection (so that we can
+//! convert back to `\r\n` on the way out as needed).
+
+use std::sync::LazyLock;
+
+use memchr::memmem;
+
+static FINDER: LazyLock<memmem::Finder> = LazyLock::new(|| memmem::Finder::new(b"\r\n"));
```

Comment:
One of the benefits of `Finder` is that you can construct one up front and reuse it, like with regexes
https://github.com/astral-sh/uv/blob/81569c47bfa91b24ff0712baf1c001ef9604676e/crates/uv-scripts/src/lib.rs#L17